lua/cfc_ulx_commands/curse/effects/sound_shuffle.lua

```diff
@@ -0,0 +1,130 @@
+local EFFECT_NAME = "SoundShuffle"
+local SOUND_DURATION_MAX = 5
+
+
+-- Create global table
+local EFFECT_NAME_LOWER = string.lower( EFFECT_NAME )
+CFCUlxCurse.EffectGlobals[EFFECT_NAME_LOWER] = CFCUlxCurse.EffectGlobals[EFFECT_NAME_LOWER] or {}
+local globals = CFCUlxCurse.EffectGlobals[EFFECT_NAME_LOWER]
+
+local soundLookup = {}
+local allSounds = nil
+local allSoundsLength = nil
+local entityMeta = FindMetaTable( "Entity" )
+
+local mathRandom = math.random
+local stringFind = string.find
+local stringSub = string.sub
+local soundDuration = globals.SoundDuration or SoundDuration
+
+
+local function getSoundForPool()
+    local attempts = 10
+    local snd = nil
+
+    while attempts > 0 do
+        snd = allSounds[mathRandom( 1, allSoundsLength )]
+
+        -- Not perfect, but covers a lot of cases.
+        local isBad =
+            stringFind( snd, "loop" ) or
+            stringSub( snd, 1, 6 ) == "synth/" or
+            soundDuration( snd ) > SOUND_DURATION_MAX
+
+        if isBad then
+            attempts = attempts - 1
+        else
+            break
+        end
+    end
+
+    return snd
+end
```

Comment:
This section gives me the creeps - keep it simple!

Also, do your consumers properly handle a case where you can't find a valid sound?
```suggestion
-- Not perfect, but covers a lot of cases.
local function isGoodSound( snd )
    if stringFind( snd, "loop" ) then return end
    if stringSub( snd, 1, 6 ) == "synth/" then return end
    if soundDuration( snd ) > SOUND_DURATION_MAX then return end
    
    return true
end

local function getSoundForPool()
    for _ = 1, 10 do
        local snd = allSounds[mathRandom( 1, allSoundsLength )]

        if isGoodSound( snd ) then
            return snd
        end
    end
end
```